lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php

```diff
@@ -401,7 +401,7 @@ private function getShortName($className)
     private function addInheritedFields(ClassMetadata $subClass, ClassMetadata $parentClass)
     {
         foreach ($parentClass->fieldMappings as $mapping) {
-            if ( ! isset($mapping['inherited']) && ! $parentClass->isMappedSuperclass) {
+            if ( ! isset($mapping['inherited']) && ! $parentClass->isMappedSuperclass && ! $parentClass->isEmbeddedClass) {
```

Comment:
Couldn't you use the `isEntity` method here? This function is used during mapping construction time, so its not critical here to avoid too many function calls.